src/GUI/HelpWindow.java

```diff
@@ -29,32 +28,33 @@ private HelpWindow() {
         JScrollPane scrollPane = new JScrollPane(helpText);
         add(scrollPane, BorderLayout.CENTER);
         pack();
-        setVisible(true);
+        setWindowSizeAndPosition();
+        this.addWindowFocusListener(new WindowFocusListener() {
+            @Override
+            public void windowGainedFocus(WindowEvent e) {
+                // Do nothing
+            }
+
+            @Override
+            public void windowLostFocus(WindowEvent e) {
+                dispose();
+            }
+        });
+        super.setVisible(true);
     }
 
-    @Override
-    public void setVisible(boolean visible) {
-        if (visible) {
-            GraphPlotterFrame frame = GraphPlotterFrame.getInstance();
-            // Determine new size based on GraphPlotterFrame.
-            Dimension frameSize = frame.getSize();
-            int newWidth = (int) (frameSize.width * 0.85); // 85% of frame width
-            int newHeight = (int) (frameSize.height * 0.85); // 85% of frame height
-            this.setSize(newWidth, newHeight);
-
-            // Calculate the new position of the HelpWindow to center it on the GraphPlotterFrame.
-            Point frameLocation = frame.getLocation();
-            int centerX = frameLocation.x + frameSize.width / 2 - newWidth / 2;
-            int centerY = frameLocation.y + frameSize.height / 2 - newHeight / 2;
-            this.setLocation(centerX, centerY);
-        }
-        super.setVisible(visible);
-    }
-
-    public static HelpWindow getInstance() {
-        if (instance == null) {
-            instance = new HelpWindow();
-        }
-        return instance;
+    private void setWindowSizeAndPosition(){
```

Comment:
just from the name, I would assume the function sets this to a position/size given by a parameter, maybe rename this to something like "centerWindow" or something

Reply:
I disagree because it doesnt just center the window it also sets the size and i dont think implies a parameter